### PR TITLE
Move more globals into _root

### PR DESCRIPTION
**Pull request by struct**
Continue moving globals we access from the hot path or in tight loops into _root for better cache locality and less implicit memcpy's of the data. This PR alone removed ~30 calls to memcpy.

### CODE REVIEW

File: utils/run_tests.sh [hunk 1]
```diff
@@ -11,11 +11,11 @@ $(ulimit -c 0)
 
 export LD_LIBRARY_PATH=build/
 
-if [[ $OSTYPE == 'darwin'* ]]; then
-    export DYLD_INSERT_LIBRARIES=build/libisoalloc.dylib
-else
-    export LD_PRELOAD=build/libisoalloc.so
-fi
+#if [[ $OSTYPE == 'darwin'* ]]; then
```
Comment (jvoisin) [inline utils/run_tests.sh:14]: This looks unrelated to this PR.

File: src/iso_alloc.c [hunk 2]
```diff
@@ -1239,8 +1238,9 @@ INTERNAL_HIDDEN iso_alloc_zone_t *iso_find_zone_range(const void *restrict p) {
 
     /* Now we check the MRU thread zone cache */
     size_t _zone_cache_count = zone_cache_count;
+    _tzc *tzc = zone_cache;
```
Comment (jvoisin) [inline src/iso_alloc.c:1241]: What is the point of doing this?

Comment (struct) [reply src/iso_alloc.c:1241]: Because `zone_cache` is a global, and a TLS global sometimes, this results in the compiler emitting calls to memcpy to make local copies of the buffer. For this specific function its 2 less than if we were to address the global directly.
